### DISCUSSION TITLE
Export Github Issues to XLSX or CSV

### DISCUSSION THREAD

**Discussion by ghost** [community/community#24634] [Apps, API and Webhooks]
<p>I’ve been trying to export all my Github issues to either Excel or as a CSV file, however, when I try to run the following command from the terminal:</p>
<pre><code>curl -i "https://api.github.com/{owner}/{repo}/issues" -u dfuentes
</code></pre>
<p>I get prompted to authenticate by entering in my password, but then I get the following returned:</p>
<pre><code>HTTP/1.1 404 Not Found
Server: GitHub.com
Date: Thu, 05 Dec 2019 17:50:38 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 87
Status: 404 Not Found
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4993
X-RateLimit-Reset: 1575571423
X-GitHub-Media-Type: github.v3; format=json
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
X-GitHub-Request-Id: D1EB:476D:5FAB09:EA04E6:5DE9436E

{
"message": "Not Found",
"documentation_url": "https://developer.github.com/v3"
}
</code></pre>
<p>Can anyone assist in helping me understand what I’m doing wrong? Or how else can I export my issues into a CSV / Excel file?</p>
<p>Thanks in advance!</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/dfuentes03">@dfuentes03</a>,</p>
<p>Thank you for being here! Unfortunately, we don’t provide the ability to export issues. This is a request that we have received in the past so I will add your request as a +1. I can’t comment on if/when such a feature might be made available, but your request will be seen by the correct team.</p>
<p>As an alternative, you can use our API to programmatically export that data:</p>
<p><a href="https://developer.github.com/v3/issues/" rel="nofollow noopener">https://developer.github.com/v3/issues/</a></p>
<p>The data is provided in JSON format.</p>
<p>We have some information about creating backups here:</p>
<p><a href="https://help.github.com/en/articles/backing-up-a-repository" rel="nofollow noopener">https://help.github.com/en/articles/backing-up-a-repository</a></p>

**Reply by mariebakke** [reply]
GitHub needs to increase the urgency of this request. It's mind boggling that this feature does not exist.

**Reply by rakeshthestorm** [reply]
Really this feature is very much needed... every now and then bosses ask for list  of pending / completed issues and even in 2024 we have to do it manually ... thats absurd

**Reply by jmopuri-Intel** [reply]
This is a must have feature that is missing! most popular bug trackers have that feature.

**Reply by nixomose** [reply]
you can get pretty far by making API calls and getting the raw data in json format and then pull out the fields you want if you need it in csv. It's not automatically, there's a little effort involved, but it is doable.

**Reply by Baherhosnijd** [reply]
I'm surprised that this has been a request 5 years earlier! This feature is very much still required

**Reply by sohelarif** [reply]
Hi @AndreaGriffiths11,
Implementing a feature to download the filtered list is not a complex task, yet it would add significant value for users. It is surprising that GitHub has not introduced this capability so far, despite several requests being raised in the past.

**This enhancement would:**
- Improve efficiency by allowing users to quickly export only the relevant subset of data.
- Reduce manual work and reliance on third-party tools or scripts.
- Provide consistency with other platforms that already offer similar functionality.
- Address one of the most frequently requested needs from the community.

Considering its high impact and relatively low implementation effort, recommend prioritizing this feature for quick action.

Best Regards,
Sohel Ansari

**Reply by nixomose** [reply]
just curious, who is "we" in the above statement?

**Comment by ghost** [comment]
<p>Thanks for your detailed response, Andrea<br>
Have a great weekend!</p>

**Comment by JanetKing61** [comment]
<p>Andrea I too would love the ability to easily export my issues into a spreadsheet to analize the requests from a macro view.  Please mark me down as a user requesting this feature.  Even if there was a print function to print the list of issues in a printer friendly mode.  Currnently I have 5 pages of issues in a project under development and I have to go page by page to print and then the format is pretty ugly-:frowning:</p>

**Comment by Joshua-rose** [comment]
<p>I have created an OAuth app for my own needs. It currently just exports the title, labels, open date, and modified date. I have opened it up so you can either fork it or make pull requests. Here is the link to the Netlify site: <a href="https://github-issues-to-excel.netlify.com/" rel="nofollow noopener">https://github-issues-to-excel.netlify.com/</a>. Here is a link to the source: <a href="https://github.com/Joshua-rose/github-issues-to-excel" rel="nofollow noopener">https://github.com/Joshua-rose/github-issues-to-excel</a>. I hope this helps a little</p>

**Comment by gavinr** [comment]
<p>Here is a command-line tool (uses the GitHub API referenced) that allows export and import of issues:</p>
<p><a href="https://github.com/gavinr/github-csv-tools" rel="nofollow noopener">https://github.com/gavinr/github-csv-tools</a></p>

**Comment by jaguardo** [comment]
<p>Thanks… I have some private repos, part of an organization that I’m a part of it… but I can’t seem to export those, only my personal repos.  any ideas? thanks!</p>

**Reply by nixomose** [reply]
in order to get at private repos you need to specify an access token with your request.

**Comment by echo-juliet** [comment]
<p>Wow it’s crazy this functionality still doesn’t exist.</p>
<p>As a PM, I would like to be able to download my team’s issues in order to triage the backlog in a more robust project management tool than Github offers (like a spreadsheet!).</p>
<p>There are integrations like Unito that will do this but it just seems like such a low-cost high-value thing to build when all that data is already in a table . . .</p>

**Comment by stolarczyk** [comment]
<p>You can also check out this one-liner that I created. Worked in my use case: <a href="https://gist.github.com/stolarczyk/1f072858d24176824dd8572741083d7d" class="inline-onebox" rel="noopener nofollow ugc">Save GitHub issues as TSV · GitHub</a></p>

**Comment by ghost** [comment]
Hi all! I would be happy if this resource for implementation will help a lot ;D +1

**Comment by jengrau** [comment]
It would be a huge help if on the Pull Request filter results page you had an export option. A non-programmer also has needs to pull this information. For our audit compliance we have to provide exports of issues to the auditor, it is very painful to take screenshots of each page of results for each repository. The auditor has also complained they cannot easily make sample selections from screenshots, especially since it's across multiple pages/files.

**Reply by faulhaber** [reply]
It would be a huge help if on the Pull Request filter results page you had an export option. A non-programmer also has needs to pull this information. 
As a non-programmer this would make my life easier.

**Comment by Louis-Shumate** [comment]
I'm surprised this isn't a feature yet. Especially since DevOps and GitHub are both owned by Microsoft now and integrate. DevOps can export a lot easier than GitHub. Seems like it would definitely be worth it and feasible.

**Comment by echo-juliet** [comment]
I don’t honestly remember how I got on this thread but I would be happy to
take on the project if GitHub has budget for a PM in this area.

**Comment by saikrishnaoggu** [comment]
Is this ticket still open ?

**Reply by jaguardo** [reply]
very much yes

**Comment by tymokvo** [comment]
Hello, I just wanted to say that this is possible if you put your issues into a table GitHub projects. 
<img width="200" src="https://github.com/community/community/assets/12144028/1c5dbbb4-c66b-4def-8ee5-c52cfd3b2e08"/>

However, I found this issue while trying to find the right place to raise a related issue that the exported format is not working quite right. The save dialog suggests that it's a CSV, but the file that is created is a TSV. On Firefox on Windows, anyway.

**Reply by mariebakke** [reply]
Also on Chrome. TSV. What the heck is a TSV file?!

**Reply by mariebakke** [reply]
This actually works! 🤯

**Reply by tymokvo** [reply]
Tab Separated Value. I transformed mine with `sed 's/\t/,/g' > file.csv` but this won't work if any of the fields use a comma `,`.

**Reply by nestor-sg** [reply]
@tymokvo thanks for your tip. This was quite helpful. Hopefully something as simple as an Export button is added in the future to any list view in GitHub. This is something so basic.

**Comment by fgonga** [comment]
2024 e ainda não temos esse recurso...

**Comment by rakeshthestorm** [comment]
Really this feature is very much needed... every now and then bosses ask for list of pending / completed issues and even in 2024 we have to do it manually ... thats absurd

**Comment by GuPuSi7ff** [comment]
Thanks for the discussion — very helpful.

I also found some useful reference notes here:
https://19wiki.net/

**Comment by sneto-scytales** [comment]
This is a must-have feature!

**Comment by echo-juliet** [comment]
Agreed. Does anyone have a repo to share?


EJetal

***
“Be patterns, be examples in all countries, places, islands, nations
wherever you come; that your carriage and life may preach among all sorts
of people, and to them; then you will come to walk cheerfully over the
world, answering that of God in everyone; whereby in them you may be a
blessing, and make the witness of God in them to bless you.”
- George Fox, Exhortation to Ministers (1656)
